src/ok.rs

```diff
@@ -58,6 +62,40 @@ impl fmt::Display for TestState {
     }
 }
 
+enum SnpIoctl {
+    Tcb,
+    Rmp,
+    Snp,
+}
+
```

Comment:
You have the right Idea, but I don't know if the name is the right one. You're testing different parameters of the SNP_PLATFORM_STATUS, which itself is an IOCTL, but you're not calling different IOCTLS. I'd rename the enum to reflect what it's actually meant for.

```suggestion
enum SnpStatusTest {
    Tcb,
    Rmp,
    Snp,
}

```